src/connectedk8s/azext_connectedk8s/tests/latest/test_connectedk8s_scenario.py

```diff
@@ -5,98 +5,521 @@
             'managed_cluster_name': managed_cluster_name
         })
 
-        self.cmd('aks create -g rohanazuregroup -n {} -s Standard_B4ms -l westeurope -c 1 --generate-ssh-keys'.format(managed_cluster_name))
-        self.cmd('aks get-credentials -g rohanazuregroup -n {managed_cluster_name} -f {kubeconfig}')
-        self.cmd('connectedk8s connect -g rohanazuregroup -n {name} -l eastus --tags foo=doo --kube-config {kubeconfig}', checks=[
+        self.cmd('aks create -g {rg} -n {managed_cluster_name} --generate-ssh-keys')
+        self.cmd('aks get-credentials -g {rg} -n {managed_cluster_name} -f {kubeconfig}')
+
+        self.cmd('connectedk8s connect -g {rg} -n {name} -l eastus --tags foo=doo --kube-config {kubeconfig} --kube-context {managed_cluster_name}', checks=[
             self.check('tags.foo', 'doo'),
             self.check('name', '{name}')
         ])
-        self.cmd('connectedk8s show -g rohanazuregroup -n {name}', checks=[
+        self.cmd('connectedk8s show -g {rg} -n {name}', checks=[
             self.check('name', '{name}'),
-            self.check('resourceGroup', 'rohanazuregroup'),
+            self.check('resourceGroup', '{rg}'),
             self.check('tags.foo', 'doo')
         ])
 
-        # Simulating the condition in which the azure-arc namespace got deleted
-        # connectedk8s delete command fails in this case
-        subprocess.run(["kubectl", "delete", "namespace", "azure-arc","--kube-config", kubeconfig])
+        os.environ.setdefault('KUBECONFIG', kubeconfig)
+        helm_client_location = install_helm_client()
+        cmd = [helm_client_location, 'get', 'values', 'azure-arc', "--namespace", "azure-arc-release", "-ojson"]
 
-        # Using the force delete command
-        # -y to supress the prompts
-        self.cmd('connectedk8s delete -g rohanazuregroup -n {name} --kube-config {kubeconfig} --force -y')
-        self.cmd('aks delete -g rohanazuregroup -n {} -y'.format(managed_cluster_name))
+        # scenario - auto-upgrade is true , so implicit upgrade commands dont work
+        self.cmd('connectedk8s update -n {name} -g {rg} --auto-upgrade true --kube-config {kubeconfig} --kube-context {managed_cluster_name}')
+        cmd_output1 = subprocess.Popen(cmd, stdout=PIPE, stderr=PIPE)
+        _, error_helm_delete = cmd_output1.communicate()
+        assert(cmd_output1.returncode == 0)
+        changed_cmd1 = json.loads(cmd_output1.communicate()[0].strip())
+        assert(changed_cmd1["systemDefaultValues"]['azureArcAgents']['autoUpdate'] == bool(1))
+
+        with self.assertRaisesRegexp(CLIError, "az connectedk8s upgrade to manually upgrade agents and extensions is only supported when auto-upgrade is set to false"):
+            self.cmd('connectedk8s upgrade -g {rg} -n {name} --kube-config {kubeconfig} --kube-context {managed_cluster_name}')
+
+        # scenario - auto upgrade is off ,then updating agnets to latest and check if the version of agents matches with latest version
```

Comment:
nit: "Turn off auto-upgrade, then updating agents .......so on". So that it is clear that we're turning off as part of this test only